multistep/commonsteps/step_download_test.go

```diff
@@ -287,6 +289,47 @@ func TestStepDownload_download(t *testing.T) {
 	os.RemoveAll(step.TargetPath)
 }
 
+func TestStepDownload_short_timeout(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Log("skipping download test on windows right now.")
+		return
+	}
+
+	os.Setenv("PACKER_GETTER_READ_TIMEOUT", "1ns")
+	prepareGetterClient()
+	defer prepareGetterClient()
```

Comment:
~~That seems weird, since you invoke it synchronously before~~
^ As I was writing this comment, I realised this was to re-initialise the client with the default values, without `PACKER_GETTER_READ_TIMEOUT`, I assume this is also why you don't use `t.Setenv` for setting the environment variable in the context of this test?
This seems OK to me if that's the rationale, but I would very much consider adding a comment here to explain that reasoning as it is not super straightforward when coming back to this code.

Note: as with the other comment I left, this can be addressed separately in a future PR, no need to leave it out of the SDK for now.